### ISSUE TITLE
L015: DISTINCT with parenthesis is not found when inside a COUNT fn

### ISSUE THREAD

**Issue by jchernjr** [sqlfluff/sqlfluff#3823]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Rule L015 (which should flag usages of `DISTINCT` keyword that looks like a function call), does not seem to find instances where it is violated, where the `DISTINCT` keyword is inside an aggregation function.

### Expected Behaviour

Expected these non-conforming `DISTINCT` usages to be flagged.

### Observed Behaviour

Linter is happy, does not report any errors.

### How to reproduce

Copy-paste sample into a file, e.g. `demo.sql`

```
SELECT count(distinct(unique_key)) AS total_count
FROM some_table
GROUP BY unique_key;
```

Run
```
sqlfluff lint --rules L015 --dialect ansi demo.sql
```

### Dialect

Tried this on Ansi and also Sparksql, but I don't think it matters.

### Version

sqlfluff, version 1.3.0

Python 3.9.13

### Configuration

not needed -- only tested one rule, as specified on command line arg

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
Interesting!

I hadn't thought about this particular application, but this does feel like a sensible extension of the rule.

I think this an _enhancement_ rather than a bug because this rule is only set up right now to look at `DISTINCT` straight after `SELECT`, rather than also within functions.

Great idea though. I think we'd need to think a bit more about how this rule crawls to make this extension.
